lib/jsonapi/serializer.ex

```diff
@@ -202,10 +207,17 @@ defmodule JSONAPI.Serializer do
   end
 
   def encode_rel_data(view, data) do
-    %{
-      type: view.type(),
-      id: view.id(data)
-    }
+    transformed_data = view.relationship_data(data)
```

Comment:
Here we will call the new hook method `relationship_data/1`, which we expect the developer to define - or else a default implementation of the method will be used (defined in view.ex - see later).